rep-0147.rst

```diff
@@ -0,0 +1,365 @@
+======================
+
+For this document we will use the following abstractions of the architecture.
+These abstractions are used to define the interfaces, however it is not required that a given system implement all the interfaces.
+A given system may implement subsets of these interfaces.
+
+
+Control Abstractions
+--------------------
+
+For controlling an aerial vehicle we will use the following terms for the interfaces.
+
+.. raw:: html
+  
+  <div class="mermaid">
+  graph LR
+    na[ ]
+    tc[Trajectory Controller]
+    ac[Attitude Controller]
+    rc[Rate Controller]
+    va[Vehicle Abstraction]
+    na ==>|Trajectory|tc
+    tc ==>|Pose|ac
+    ac ==>|Rate|rc
+    rc ==>|Acceleration|va
+    
+    va -.-> am[Allocation Matrix]
+    va -.-> mixer
+    
+    am -.-> Actuators
+    mixer -.-> Actuators 
+  </div>
+
+High Level Architecture
+-----------------------
+
+A higher level view of the system uses the following abstraction.
+
+.. raw:: html
+  
+  <div class="mermaid">
+  graph LR
+    GCS[Ground Control Station]
+    CS[Collision Sensors]
+    OS[Odometry Sensors]
+    C[Controller]
+    L[ ]
+    style L fill:#fff,stroke:#ff0,stroke-width:0px;
+    L --> CS
+    L --> GCS
+    L --> OS
+    subgraph Generic Planning Framework
+    CM[Collision Mapping]
+    MSC[Minimum Snap Controller]
+    CM ---|Shared State|MSC
+    end
+    CS -->|Obstacle Observations|CM
+    OS -->|Odometry|MSC
+    OS -->|Odometry|CM
+    OS -->|Odometry|C
+    GCS -->|Goals|MSC
+    MSC -->|Trajectory|C
+    %% Links to L 
+    linkStyle 0 fill:#fff,stroke:#ff0,stroke-width:0px;
+    linkStyle 1 fill:#fff,stroke:#ff0,stroke-width:0px;
+    linkStyle 2 fill:#fff,stroke:#ff0,stroke-width:0px;
+  </div>
+
+
+Coordinate Frames
+=================
+
+It is expected that users of this interface will also follow the coordinate frame conventions layed out in REP 105 [#rep105]_.
+
+Proposed Interfaces
+===================
+
+Control Interfaces
+------------------
+
+Trajectory Control Interface
+,,,,,,,,,,,,,,,,,,,,,,,,,,,,
+
+For sending trajectories in 3D space there already exists a message used for free space planning.
+This message used is ``trajectory_msgs/MultiDOFJointTrajectory`` [#multidoftraj]_ and is actively used by planning frameworks already.
+
+This message is not the most straight forward to parse as it has a lot of optional fields which increases the complexity and requires consistent labelling and filling.
+The complexity of the message parsing is traded off against supporting a relatively compact representations.
+
+* Datatype: ``trajectory_msgs/MultiDOFJointTrajectory`` [#multidoftraj]_ 
+* Topic: ``command_trajectory``
+
+Pose Interface
+,,,,,,,,,,,,,,
+
+The message to send a goal pose is ``geometry_msgs/PoseStamped`` [#posestamped]_.
+This is a very highly used standard message and contains precisely the information needed.
+
+* Datatype: ``geometry_msgs/PoseStamped`` [#posestamped]_
+* Topic: ``command_pose``
+
+
+Rate Interface
```

Comment:
I picked rate for this since typically if you hold a constant attitude it will trend toward a constant velocity.  This also gives a relatively clear progression, that there's position -> rate -> acceleration interfaces which are each derivatives of each other. The expression of each we've stuck with aircraft expressions instead of using the pure geometric formats since that's how most people think about flight controls.  But if the relationship is clearly documented I'm not against renaming.